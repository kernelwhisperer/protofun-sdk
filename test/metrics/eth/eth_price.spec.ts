import { expect, it } from "vitest"

import query from "../../../src/metrics/eth/eth_price"

it("Daily candles", async () => {
  // act
  const candles = await query({
    timeframe: "Day",
    since: "0",
    until: "1503273600",
  })
  // assert
  expect(candles).toMatchSnapshot()
})

it("Daily candles - EIP 1559", async () => {
  // act
  const candles = await query({
    timeframe: "Day",
    since: "1628121600",
    until: "1628553600",
  })
  // assert
  expect(candles).toMatchSnapshot()
})

it("Hourly candles", async () => {
  // act
  const candles = await query({
    timeframe: "Hour",
    since: "0",
    until: "1502960400",
  })
  // assert
  expect(candles).toMatchSnapshot()
})

it("Block candles", async () => {
  // act
  try {
    await query({
      timeframe: "Block",
    })
  } catch (error) {
    // assert
    expect(error).toMatchInlineSnapshot(
      "[Error: Timeframe 'Block' is not supported for this metric.]"
    )
  }
})
