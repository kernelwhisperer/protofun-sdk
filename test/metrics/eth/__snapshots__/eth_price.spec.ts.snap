// Vitest Snapshot v1

exports[`Daily candles - EIP 1559 1`] = `
[
  {
    "close": "2826.99000000",
    "high": "2843.98000000",
    "low": "2529.91000000",
    "open": "2724.55000000",
    "timestamp": "1628121600",
  },
  {
    "close": "2891.58000000",
    "high": "2946.75000000",
    "low": "2720.06000000",
    "open": "2827.01000000",
    "timestamp": "1628208000",
  },
  {
    "close": "3160.31000000",
    "high": "3171.00000000",
    "low": "2865.50000000",
    "open": "2891.54000000",
    "timestamp": "1628294400",
  },
  {
    "close": "3011.85000000",
    "high": "3189.99000000",
    "low": "2949.03000000",
    "open": "3160.48000000",
    "timestamp": "1628380800",
  },
  {
    "close": "3160.52000000",
    "high": "3186.00000000",
    "low": "2893.23000000",
    "open": "3011.84000000",
    "timestamp": "1628467200",
  },
  {
    "close": "3141.66000000",
    "high": "3233.00000000",
    "low": "3051.22000000",
    "open": "3160.51000000",
    "timestamp": "1628553600",
  },
]
`;

exports[`Daily candles 1`] = `
[
  {
    "close": "302.00000000",
    "high": "312.18000000",
    "low": "298.00000000",
    "open": "301.13000000",
    "timestamp": "1502928000",
  },
  {
    "close": "293.96000000",
    "high": "311.79000000",
    "low": "283.94000000",
    "open": "302.00000000",
    "timestamp": "1503014400",
  },
  {
    "close": "290.91000000",
    "high": "299.90000000",
    "low": "278.00000000",
    "open": "293.31000000",
    "timestamp": "1503100800",
  },
  {
    "close": "299.10000000",
    "high": "300.53000000",
    "low": "282.85000000",
    "open": "289.41000000",
    "timestamp": "1503187200",
  },
  {
    "close": "323.29000000",
    "high": "346.52000000",
    "low": "294.60000000",
    "open": "299.10000000",
    "timestamp": "1503273600",
  },
]
`;

exports[`Hourly candles 1`] = `
[
  {
    "close": "301.61000000",
    "high": "302.57000000",
    "low": "298.00000000",
    "open": "301.13000000",
    "timestamp": "1502942400",
  },
  {
    "close": "303.10000000",
    "high": "303.28000000",
    "low": "300.00000000",
    "open": "301.61000000",
    "timestamp": "1502946000",
  },
  {
    "close": "302.68000000",
    "high": "304.44000000",
    "low": "301.90000000",
    "open": "302.40000000",
    "timestamp": "1502949600",
  },
  {
    "close": "307.96000000",
    "high": "307.96000000",
    "low": "302.60000000",
    "open": "302.68000000",
    "timestamp": "1502953200",
  },
  {
    "close": "308.62000000",
    "high": "309.97000000",
    "low": "307.00000000",
    "open": "307.95000000",
    "timestamp": "1502956800",
  },
  {
    "close": "310.00000000",
    "high": "312.00000000",
    "low": "308.62000000",
    "open": "308.62000000",
    "timestamp": "1502960400",
  },
]
`;
